src/main/java/luffy/Luffy.java

```diff
@@ -0,0 +1,68 @@
+package luffy;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.util.regex.Pattern;
+
+/**
+ * Luffy is a todolist program. Built for CS2103T Individual Project 2022 S1.
+ *
+ * @author Silas Tay A0233425M
+ */
+public class Luffy {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    private static final String FILE_PATH = "./data/data.txt";
+
+    /**
+     * Construtor for Luffy object.
+     *
+     * @param filePath File path to data file
+     */
+    public Luffy(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        try {
+            this.tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            this.ui.showLoadingError();
+            this.tasks = new TaskList();
```

Comment:
Perhaps can come up with alternative to handle the logic here? In general, not the best practice to use `try-catch` block for control flow.